### ISSUE TITLE
Few docset properties (for Zealdocs/Dash) shoud be changed

### ISSUE THREAD

**Issue by NEK-RA** [flutter/flutter#97386]
**First of all:** I doesn't build docset by myself, I'm using this one - https://api.flutter.dev/offline/flutter.xml
**Description:**
At the path `.../flutter.docset/Contents` there is an `Info.plist` file which contains:
 ```
...
<key>dashIndexFilePath</key>
<string>index.html</string>
...
```
I have no MacOS device, so I don't know how it works in Dash (correct or not), but at Zealdocs (Windows) there is a problem that `dashIndexFilePath` is relative to `flutter.docset\Contents\Resources\Documents` but actually this `Documents` folder contains only `doc` folder, where all the docset files are stored.

As result - all works from the search, but if you click on docset from the list, you'll see the error of opening this `index.html` as mentioned at https://github.com/flutter/flutter/issues/51446

Workarounds to solve the problems are:
- move all files from `doc` to `Documents`, but this breaks the search until manually alter all `flutter.docset/Contents/Resources/docSet.dsidx` entries to remove `doc/` from all the paths (each docset which I have contain root at `Documents`, and not in any nested folder)
- update `Info.plist` for the `dashIndexFilePath` to have value `doc/index.html` instead of `index.html` (the easier one)

Also I wonder, why Flutter use `DocSetPlatformFamily` as `dartlang` and not `flutter` like any other framework docset? 

I don't know for the others, but when I just downloaded the flutter docset for the first time, I thought that docset is broken due to this `doc` folder issue and no keyword for the scoped search (only today I got that flutter docset uses the `dartlang` keyword) 😮

**Comment by chinmaygarde** [comment]
You're right that the `dashIndexFilePath` entry is incorrect. On Dash on Mac, I am no longer able to select a docset to view its index page. I think that is no longer a feature in the newer versions of Dash and so no one noticed the issue. The fix ought to be easy enough. Just change `index.html` to `doc/index.html` in the template. As you say, browsing using search is fine.

@gspencergoog I believe you wired up docset generation? Can you point us to the template file used to generate the Info.plist?

**Comment by maks** [comment]
On the topic of Docset fixes, https://github.com/flutter/flutter/issues/25104 seems to have regressed and using the feed from https://api.flutter.dev/offline/flutter.xml the icon again no longer shows for me using Zeal, so perhaps could be fixed at the same time as this is?

![Screenshot from 2022-09-21 09-41-36](https://user-images.githubusercontent.com/71999/191384094-b89926f5-c5c1-49ee-b6d2-5d6beb042fcc.png)

**Comment by NEK-RA** [comment]
@maks actually any icon placed into `flutter.docset` folder as `icon.png` seems to work fine. 
I do understand that it makes no sence until it won't be fixed in generation tools/configs, but let it be there for those who wanna see an icon anyway

**Comment by maks** [comment]
@NEK-RA thanks for the tip about manually placing the icon in the `$HOME/.local/share/Zeal/Zeal/docsets/flutter.docset/` folder, though one downside I noticed with this workaround is it will get overwritten on each docset update, so would be nice if this could be fixed on the Flutter side, @gspencergoog any chance you could fix it again like you did in #25104 🙏🏻 .
